webrtc.html

```diff
@@ -8736,10 +8736,11 @@ <h2>Attributes</h2>
               whether the user agent is able to transmit text asynchronously
               with script execution.) This does not include framing overhead
               incurred by the protocol, or buffering done by the operating
-              system or network hardware. If the channel is closed, this
-              attribute's value will only increase with each call to the
-              <code><a data-for="RTCDataChannel">send()</a></code> method (the
-              attribute does not reset to zero once the channel closes).</p>
+              system or network hardware. This attribute's value will only
+              increase with each call to the <code><a data-for=
+              "RTCDataChannel">send()</a></code> method as long as the <a>
+              [[\ReadyState]]</a> slot is <code>open</code>. However, the
+              attribute does not reset to zero once the channel closes.</p>
```

Comment:
Does this only affect non graceful closures, or closures that race with data being sent?

Reply:
I'd say both and this is my interpretation:
* If the data channel closes abruptly, the value remains the same and does not reset because the state goes directly from whatever to `closed`. I believe that implementation could flush the underlying buffer. They just need to make sure the value does not reset.
* If the data channel closes gracefully, the value will be `0` once the channel reaches the `closed` state. Before that, all pending data has been written during the `closing` state. Even though the value decreases, I wouldn't call this a *reset*.

I agree that the description isn't crystal clear in that regard. Suggestions welcome.

Reply:
By the way, graceful shutdown of data channels isn't something that's really specified right now in the W3C spec (even though it is specified in [draft-ietf-rtcweb-data-channel-13](https://tools.ietf.org/html/draft-ietf-rtcweb-data-channel-13#section-6.7)). See for example [this bug in Firefox](https://bugzilla.mozilla.org/show_bug.cgi?id=1413198). I guess I just forgot to open an issue (once again).

Reply:
Suggestion:

> During the `closing` state, the attribute's value may decrease until the underlying buffer is completely sent. However, the attribute's value is frozen once the channel reaches the `closed` state.

Reply:
Your suggestion, and your interpretation of graceful/non-graceful closing behavior, sounds good to me. Though I thought we were ok with regards to graceful closure? "An RTCDataChannel object's underlying data transport may be torn down in a non-abrupt manner by running the closing procedure."

Reply:
Ok, I'll update the description in the PR.

I'll open a new issue for the closing procedure stuff and will mention you.